github-actions/src/main/scala/org/typelevel/sbt/gha/WorkflowJob.scala

```diff
@@ -107,7 +112,54 @@ object WorkflowJob {
       container,
       environment,
       concurrency,
-      timeoutMinutes
+      timeoutMinutes,
+      outputs
+    )
+
+  // Kept for binary compatibility. It can be removed in a future major release.
+  def apply(
+      id: String,
+      name: String,
+      steps: List[WorkflowStep],
+      sbtStepPreamble: List[String],
+      cond: Option[String],
+      permissions: Option[Permissions],
+      env: Map[String, String],
+      oses: List[String],
+      scalas: List[String],
+      javas: List[JavaSpec],
+      needs: List[String],
+      matrixFailFast: Option[Boolean],
+      matrixAdds: Map[String, List[String]],
+      matrixIncs: List[MatrixInclude],
+      matrixExcs: List[MatrixExclude],
+      runsOnExtraLabels: List[String],
+      container: Option[JobContainer],
+      environment: Option[JobEnvironment],
+      concurrency: Option[Concurrency],
+      timeoutMinutes: Option[Int]): WorkflowJob =
+    apply(
+      id,
+      name,
+      steps,
+      sbtStepPreamble,
+      cond,
+      permissions,
+      env,
+      oses,
+      scalas,
+      javas,
+      needs,
+      matrixFailFast,
+      matrixAdds,
+      matrixIncs,
+      matrixExcs,
+      runsOnExtraLabels,
+      container,
+      environment,
+      concurrency,
+      timeoutMinutes,
+      Map()
     )
 
   private final case class Impl(
```

Comment:
We don't need to preserve the binary-compatibility of `Impl` because it is a private implementation class. So most of the changes below are unnecessary. You may need to add MiMa filters.